### PR TITLE
add kbdfans/maja rgb keyboard

### PR DESCRIPTION
**Pull request by moyi4681**
add kbdfans/maja rgb keyboard

### CODE REVIEW

File: keyboards/kbdfans/maja/keymaps/default/keymap.c [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+			CTL_T(KC_CAPS),KC_A,     KC_S,     KC_D,  KC_F,  KC_G,  KC_H,    KC_J,  KC_K,    KC_L,     KC_SCLN,  KC_QUOT,           KC_ENT,    KC_PGDN,\
+		    KC_LSFT,       KC_Z,     KC_X,     KC_C,  KC_V,  KC_B,  KC_N,    KC_M,  KC_COMM, KC_DOT,   KC_SLSH,  KC_RSFT,  KC_UP,   KC_END,\
+		    KC_LCTL,                 KC_LGUI,  KC_LALT,      KC_SPC,         KC_RALT,        KC_RCTL,                      KC_LEFT, KC_DOWN,   KC_RIGHT),
+		[1] = LAYOUT( /* FN */
+			KC_GESC,       KC_F1,    KC_F2,   KC_F3,  KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,   KC_F11,  KC_F12,  RESET,   KC_HOME,\
+		    KC_TRNS,       RGB_TOG,  RGB_MOD, RGB_HUI,RGB_HUD, RGB_SAI, RGB_SAD, RGB_VAI, RGB_VAD, KC_TRNS, KC_PSCR, KC_SLCK,  KC_PAUS, RESET,    KC_PGUP,\
+			CTL_T(KC_CAPS),RGB_SPI,  RGB_SPD, KC_TRNS,KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,  KC_TRNS,          EEP_RST,  KC_PGDN,\
+		    KC_LSFT,       KC_TRNS,  KC_TRNS, KC_TRNS,KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,  KC_TRNS, KC_VOLU, KC_MUTE,\
+		    KC_TRNS,                 KC_TRNS, KC_TRNS,         KC_TRNS,           KC_TRNS,         KC_TRNS,                    KC_MPRV, KC_VOLD,  KC_MNXT),
```
Comment (fauxpark) [inline keyboards/kbdfans/maja/keymaps/default/keymap.c:14]:
```suggestion
			KC_GESC,       KC_1,     KC_2,     KC_3,  KC_4,  KC_5,  KC_6,    KC_7,  KC_8,    KC_9,     KC_0,     KC_MINS,  KC_EQL,  KC_BSPC,   KC_HOME,
		    KC_TAB,        KC_Q,     KC_W,     KC_E,  KC_R,  KC_T,  KC_Y,    KC_U,  KC_I,    KC_O,     KC_P,     KC_LBRC,  KC_RBRC, KC_BSLASH, KC_PGUP,
			CTL_T(KC_CAPS),KC_A,     KC_S,     KC_D,  KC_F,  KC_G,  KC_H,    KC_J,  KC_K,    KC_L,     KC_SCLN,  KC_QUOT,           KC_ENT,    KC_PGDN,
		    KC_LSFT,       KC_Z,     KC_X,     KC_C,  KC_V,  KC_B,  KC_N,    KC_M,  KC_COMM, KC_DOT,   KC_SLSH,  KC_RSFT,  KC_UP,   KC_END,
		    KC_LCTL,                 KC_LGUI,  KC_LALT,      KC_SPC,         KC_RALT,        KC_RCTL,                      KC_LEFT, KC_DOWN,   KC_RIGHT),
		[1] = LAYOUT( /* FN */
			KC_GESC,       KC_F1,    KC_F2,   KC_F3,  KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,   KC_F11,  KC_F12,  RESET,   KC_HOME,
		    KC_TRNS,       RGB_TOG,  RGB_MOD, RGB_HUI,RGB_HUD, RGB_SAI, RGB_SAD, RGB_VAI, RGB_VAD, KC_TRNS, KC_PSCR, KC_SLCK,  KC_PAUS, RESET,    KC_PGUP,
			CTL_T(KC_CAPS),RGB_SPI,  RGB_SPD, KC_TRNS,KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,  KC_TRNS,          EEP_RST,  KC_PGDN,
		    KC_LSFT,       KC_TRNS,  KC_TRNS, KC_TRNS,KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,  KC_TRNS, KC_VOLU, KC_MUTE,
		    KC_TRNS,                 KC_TRNS, KC_TRNS,         KC_TRNS,           KC_TRNS,         KC_TRNS,                    KC_MPRV, KC_VOLD,  KC_MNXT),
```

File: keyboards/kbdfans/maja/keymaps/default/keymap.c [hunk 2]
```diff
@@ -0,0 +1,15 @@
+#include QMK_KEYBOARD_H
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
```
Comment (fauxpark) [inline keyboards/kbdfans/maja/keymaps/default/keymap.c:2]:
```suggestion

const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
```

File: keyboards/kbdfans/maja/keymaps/via/keymap.c [hunk 3]
```diff
@@ -0,0 +1,15 @@
+#include QMK_KEYBOARD_H
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
```
Comment (fauxpark) [inline keyboards/kbdfans/maja/keymaps/via/keymap.c:2]:
```suggestion

const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
```

File: keyboards/kbdfans/maja/keymaps/via/keymap.c [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+			CTL_T(KC_CAPS),KC_A,     KC_S,     KC_D,  KC_F,  KC_G,  KC_H,    KC_J,  KC_K,    KC_L,     KC_SCLN,  KC_QUOT,           KC_ENT,    KC_PGDN,\
+		    KC_LSFT,       KC_Z,     KC_X,     KC_C,  KC_V,  KC_B,  KC_N,    KC_M,  KC_COMM, KC_DOT,   KC_SLSH,  KC_RSFT,  KC_UP,   KC_END,\
+		    KC_LCTL,                 KC_LGUI,  KC_LALT,      KC_SPC,         KC_RALT,        KC_RCTL,                      KC_LEFT, KC_DOWN,   KC_RIGHT),
+		[1] = LAYOUT( /* FN */
+			KC_GESC,       KC_F1,    KC_F2,   KC_F3,  KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,   KC_F11,  KC_F12,  RESET,   KC_HOME,\
+		    KC_TRNS,       RGB_TOG,  RGB_MOD, RGB_HUI,RGB_HUD, RGB_SAI, RGB_SAD, RGB_VAI, RGB_VAD, KC_TRNS, KC_PSCR, KC_SLCK,  KC_PAUS, RESET,    KC_PGUP,\
+			CTL_T(KC_CAPS),RGB_SPI,  RGB_SPD, KC_TRNS,KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,  KC_TRNS,          EEP_RST,  KC_PGDN,\
+		    KC_LSFT,       KC_TRNS,  KC_TRNS, KC_TRNS,KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,  KC_TRNS, KC_VOLU, KC_MUTE,\
+		    KC_TRNS,                 KC_TRNS, KC_TRNS,         KC_TRNS,           KC_TRNS,         KC_TRNS,                    KC_MPRV, KC_VOLD,  KC_MNXT),
```
Comment (fauxpark) [inline keyboards/kbdfans/maja/keymaps/via/keymap.c:14]:
```suggestion
			KC_GESC,       KC_1,     KC_2,     KC_3,  KC_4,  KC_5,  KC_6,    KC_7,  KC_8,    KC_9,     KC_0,     KC_MINS,  KC_EQL,  KC_BSPC,   KC_HOME,
		    KC_TAB,        KC_Q,     KC_W,     KC_E,  KC_R,  KC_T,  KC_Y,    KC_U,  KC_I,    KC_O,     KC_P,     KC_LBRC,  KC_RBRC, KC_BSLASH, KC_PGUP,
			CTL_T(KC_CAPS),KC_A,     KC_S,     KC_D,  KC_F,  KC_G,  KC_H,    KC_J,  KC_K,    KC_L,     KC_SCLN,  KC_QUOT,           KC_ENT,    KC_PGDN,
		    KC_LSFT,       KC_Z,     KC_X,     KC_C,  KC_V,  KC_B,  KC_N,    KC_M,  KC_COMM, KC_DOT,   KC_SLSH,  KC_RSFT,  KC_UP,   KC_END,
		    KC_LCTL,                 KC_LGUI,  KC_LALT,      KC_SPC,         KC_RALT,        KC_RCTL,                      KC_LEFT, KC_DOWN,   KC_RIGHT),
		[1] = LAYOUT( /* FN */
			KC_GESC,       KC_F1,    KC_F2,   KC_F3,  KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,   KC_F11,  KC_F12,  RESET,   KC_HOME,
		    KC_TRNS,       RGB_TOG,  RGB_MOD, RGB_HUI,RGB_HUD, RGB_SAI, RGB_SAD, RGB_VAI, RGB_VAD, KC_TRNS, KC_PSCR, KC_SLCK,  KC_PAUS, RESET,    KC_PGUP,
			CTL_T(KC_CAPS),RGB_SPI,  RGB_SPD, KC_TRNS,KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,  KC_TRNS,          EEP_RST,  KC_PGDN,
		    KC_LSFT,       KC_TRNS,  KC_TRNS, KC_TRNS,KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,  KC_TRNS, KC_VOLU, KC_MUTE,
		    KC_TRNS,                 KC_TRNS, KC_TRNS,         KC_TRNS,           KC_TRNS,         KC_TRNS,                    KC_MPRV, KC_VOLD,  KC_MNXT),
```

File: keyboards/kbdfans/maja/maja.c [hunk 5]
```diff
@@ -0,0 +1,114 @@
+#include "maja.h"
+const is31_led g_is31_leds[DRIVER_LED_TOTAL] = {
```
Comment (fauxpark) [inline keyboards/kbdfans/maja/maja.c:2]:
```suggestion

const is31_led g_is31_leds[DRIVER_LED_TOTAL] = {
```

File: keyboards/kbdfans/maja/maja.c [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+
+void matrix_init_kb(void) {
+    matrix_init_user();
+}
+void matrix_scan_kb(void) {
+    matrix_scan_user();
+}
+bool process_record_kb(uint16_t keycode, keyrecord_t *record) {
+    return process_record_user(keycode, record);
+}
```
Comment (fauxpark) [inline keyboards/kbdfans/maja/maja.c:104]:
```suggestion
```

File: keyboards/kbdfans/maja/maja.c [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+void suspend_power_down_kb(void)
+{
+    rgb_matrix_set_suspend_state(true);
+    suspend_power_down_user();
+}
+void suspend_wakeup_init_kb(void)
+{
```
Comment (fauxpark) [inline keyboards/kbdfans/maja/maja.c:111]:
```suggestion
void suspend_power_down_kb(void) {
    rgb_matrix_set_suspend_state(true);
    suspend_power_down_user();
}
void suspend_wakeup_init_kb(void) {
```

File: keyboards/kbdfans/maja/maja.h [hunk 8]
```diff
@@ -0,0 +1,16 @@
+#pragma once
+#include "quantum.h"
+#define XXX KC_NO
+#define LAYOUT( \
```
Comment (fauxpark) [inline keyboards/kbdfans/maja/maja.h:4]:
```suggestion

#include "quantum.h"

#define XXX KC_NO

#define LAYOUT( \
```

File: keyboards/kbdfans/maja/readme.md [hunk 9]
```diff
@@ -0,0 +1,13 @@
+# MAJA
+
+![MAJA](imgur.com image replace me!)
```
Comment (fauxpark) [inline keyboards/kbdfans/maja/readme.md:3]: Please put an image here :)

File: keyboards/kbdfans/maja/readme.md [hunk 10]
```diff
@@ -0,0 +1,13 @@
...
+
+Keyboard Maintainer: [DZTECH](https://github.com/moyi4681)    
+Hardware Supported: MAJA   
+Hardware Availability: [KBDFans](https://kbdfans.cn/)
```
Comment (fauxpark) [inline keyboards/kbdfans/maja/readme.md:7]:
```suggestion
* Keyboard Maintainer: [DZTECH](https://github.com/moyi4681)
* Hardware Supported: MAJA
* Hardware Availability: [KBDFans](https://kbdfans.cn/)
```

File: keyboards/kbdfans/maja/keymaps/via/keymap.c [hunk 11]
```diff
@@ -0,0 +1,16 @@
...
+			CTL_T(KC_CAPS),KC_A,     KC_S,     KC_D,  KC_F,  KC_G,  KC_H,    KC_J,  KC_K,    KC_L,     KC_SCLN,  KC_QUOT,           KC_ENT,    KC_PGDN,
+		    KC_LSFT,       KC_Z,     KC_X,     KC_C,  KC_V,  KC_B,  KC_N,    KC_M,  KC_COMM, KC_DOT,   KC_SLSH,  KC_RSFT,  KC_UP,   KC_END,
+		    KC_LCTL,                 KC_LGUI,  KC_LALT,      KC_SPC,         KC_RALT,        KC_RCTL,                      KC_LEFT, KC_DOWN,   KC_RIGHT),
+		[1] = LAYOUT( /* FN */
+			KC_GESC,       KC_F1,    KC_F2,   KC_F3,  KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,   KC_F11,  KC_F12,  RESET,   KC_HOME,
+		    KC_TRNS,       RGB_TOG,  RGB_MOD, RGB_HUI,RGB_HUD, RGB_SAI, RGB_SAD, RGB_VAI, RGB_VAD, KC_TRNS, KC_PSCR, KC_SLCK,  KC_PAUS, RESET,    KC_PGUP,
+			CTL_T(KC_CAPS),RGB_SPI,  RGB_SPD, KC_TRNS,KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,  KC_TRNS,          EEP_RST,  KC_PGDN,
+		    KC_LSFT,       KC_TRNS,  KC_TRNS, KC_TRNS,KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,  KC_TRNS, KC_VOLU, KC_MUTE,
+		    KC_TRNS,                 KC_TRNS, KC_TRNS,         KC_TRNS,           KC_TRNS,         KC_TRNS,                    KC_MPRV, KC_VOLD,  KC_MNXT),
+ };
```
Comment (noroadsleft) [inline keyboards/kbdfans/maja/keymaps/via/keymap.c:16]:
```suggestion
		[2] = LAYOUT(
			_______,       _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
		    _______,       _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
			_______,       _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______, _______,
		    _______,       _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
		    _______,                _______, _______,          _______,          _______,          _______,                   _______, _______, _______),
		[3] = LAYOUT(
			_______,       _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
		    _______,       _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
			_______,       _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______, _______,
		    _______,       _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
		    _______,                _______, _______,          _______,          _______,          _______,                   _______, _______, _______),
 };
```

File: keyboards/kbdfans/maja/rules.mk [hunk 12]
```diff
@@ -0,0 +1,33 @@
...
+BOOTLOADER = atmel-dfu
+
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no        # Commands for debug and configuration
```
Comment (noroadsleft) [inline keyboards/kbdfans/maja/rules.mk:21]:
```suggestion
COMMAND_ENABLE = no         # Commands for debug and configuration
```

File: keyboards/kbdfans/maja/rules.mk [hunk 13]
```diff
@@ -0,0 +1,33 @@
...
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no      # Enable keyboard backlight functionality
```
Comment (noroadsleft) [inline keyboards/kbdfans/maja/rules.mk:26]:
```suggestion
NKRO_ENABLE = yes           # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
```

File: keyboards/kbdfans/maja/rules.mk [hunk 14]
```diff
@@ -0,0 +1,33 @@
...
+MIDI_ENABLE = no            # MIDI support
+UNICODE_ENABLE = no         # Unicode
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no           # Audio output on port C6
+FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
+RGB_MATRIX_ENABLE = yes     # Use RGB matrix
+NO_USB_STARTUP_CHECK = no          # Disable initialization only when usb is plugged in
```
Comment (noroadsleft) [inline keyboards/kbdfans/maja/rules.mk:33]:
```suggestion
NO_USB_STARTUP_CHECK = no   # Disable initialization only when usb is plugged in
```

File: keyboards/kbdfans/maja/config.h [hunk 15]
```diff
@@ -0,0 +1,44 @@
+#pragma once
+#include "config_common.h"
```
Comment (fauxpark) [inline keyboards/kbdfans/maja/config.h:2]:
```suggestion
#pragma once

#include "config_common.h"

```

File: keyboards/kbdfans/maja/readme.md [hunk 16]
```diff
@@ -0,0 +1,13 @@
+# MAJA
+
+![MAJA](https://i.ibb.co/ZxWRgg3/2.png)
+![MAJA](https://i.ibb.co/kgXQkgF/1.png)
+* Keyboard Maintainer: [DZTECH](https://github.com/moyi4681)
```
Comment (fauxpark) [inline keyboards/kbdfans/maja/readme.md:5]:
```suggestion

* Keyboard Maintainer: [DZTECH](https://github.com/moyi4681)
```

**Comment by noroadsleft** [comment]
Thanks!

**Pull request merged**